### PR TITLE
Update Ethermint app/ Daemon to run functional TM node

### PR DESCRIPTION
**Pull request by austinabell**
Updates the app framework to get the tendermint node daemon commands to function #67 and completes the functionality missing from #6 (#66). Although it seems as if this code is excessive, it contains only the base modules for a functional chain. 

The Cosmos SDK had been updated a lot since the creation of Ethermint and this update reflects the changes and updates the structure to the up to date instantiation of a Cosmos SDK app. These changes allow the EVM module to be added on top of the application so it can be easily connected and tested with this functional node.

To run the node, I ran these commands:
```
make install && rm -rf ~/.emint* && emintd init moniker --chain-id testchain && emintcli config chain-id testchain && emintcli config output json && emintcli config indent true && emintcli config trust-node true && emintcli keys add austin && emintd add-genesis-account $(emintcli keys show austin -a) 1000photon,100000000stake && emintd gentx --name austin && emintd collect-gentxs && emintd validate-genesis && emintd start
```
Ignore the couldn't connect to peers error as those won't be configured but blocks will still be committed. If you're lazy and don't want to type in a passphrase 3 times each run hopefully this will handle the prompts in your shell:
```
make install 
rm -rf ~/.emint*
emintd init moniker --chain-id testchain
emintcli config chain-id testchain
emintcli config output json
emintcli config indent true
emintcli config trust-node true
emintcli keys add austin
testpass
testpass
emintd add-genesis-account $(emintcli keys show austin -a) 1000photon,100000000stake
emintd gentx --name austin
testpass
emintd collect-gentxs
emintd validate-genesis
emintd start
```

Edit: The only module I think that could maybe be subbed out may be the governance module, but we will need to clarify if that should be included. All others seem necessary to have a functional and usable zone. Will have to clarify if Cosmos governance will be included in Ethermint. 

Also, this PR allows #58 to be implemented.

### CODE REVIEW

File: app/ethermint.go [hunk 1]
```diff
@@ -1,74 +1,73 @@
...
-	storeKeySupply      = sdk.NewKVStoreKey("supply")
-	storeKeyParams      = sdk.NewKVStoreKey("params")
-	storeKeyTransParams = sdk.NewTransientStoreKey("transient_params")
 )
 
 // MakeCodec generates the necessary codecs for Amino
 func MakeCodec() *codec.Codec {
 	var cdc = codec.New()
+
+	// TODO: Move this codec to module (Issue #12 https://github.com/ChainSafe/ethermint/issues/12)
```
Comment (ansermino) [inline app/ethermint.go:68]: I think this issue can be closed, we call the other function on the next line

File: app/ethermint.go [hunk 2]
```diff
@@ -141,88 +168,125 @@ func NewEthermintApp(logger tmlog.Logger, db dbm.DB, baseAppOpts ...func(*bam.Ba
...
-	_ sdk.Context, req abci.RequestInitChain,
-) abci.ResponseInitChain {
+// ExportAppStateAndValidators exports the state of the application for a genesis
+// file.
+func (app *EthermintApp) ExportAppStateAndValidators(forZeroHeight bool, jailWhiteList []string,
+) (appState json.RawMessage, validators []tmtypes.GenesisValidator, err error) {
 
-	var genesisState evm.GenesisState
-	stateJSON := req.AppStateBytes
+	// as if they could withdraw from the start of the next block
```
Comment (ansermino) [inline app/ethermint.go]: Suspect there is some useful info here, not sure what this means though

**Review by ansermino — DISMISSED** [review]
:+1:

File: app/ethermint.go (hunk 1 above)
Comment (austinabell) [reply app/ethermint.go:68]: Updated issue description, resolved in person

**Review by ansermino — APPROVED** [review]

**Pull request merged**